doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,806 @@
+
+    Returns:
+        An integer, the number of modules available on this chassis
+```
+
+get_all_modules(self):
+```
+    Retrieves all modules available on this chassis including DPUs
+
+    Returns:
+        A list of objects derived from ModuleBase representing all modules
+        available on this chassis
+```
+
+get_module(self, index):
+```
+    Retrieves module represented by index <index> switch:0, DPU1:1 and so on
+
+    Args:
+        index: An integer, the index of the module to retrieve
+
+    Returns:
+        An object derived from ModuleBase representing the specified module
+```
+
+get_module_index(self, module_name):
+```
+    Retrieves module index from the module name
+
+    Args:
+        module_name: A string, Ex. SWITCH, DPU1, DPU2 ... DPUX
+
+    Returns:
+        An integer, the index of the ModuleBase object in the module_list
+```
+#### 3.1.2 ChassisBase class new APIs
+is_smartswitch(self):
+```
+    Retrieves whether the sonic instance is part of smartswitch
+
+    Returns:
+      True
+```
+
+get_module_dpu_port(self, index):
+```
+    Retrieves the DPU port - internal ASIC port for DPU represented by
+    DPU index. Platforms that require to overwrite the platform.json file
+    will use this API
+
+    This valid only on the Switch and not on DPUs
+
+    Args:
+        index: An integer, the index of the module to retrieve
+
+    Returns:
+        DPU Port: A string Ex: For index:0 "dpu0", index:1 "dpu1"
+        See the NPU to DPU port mapping
+```
+#### 3.1.3 NPU to DPU port mapping
+platform.json of NPU/switch will show the NPU port to DPU port mapping. This will be used by services early in the system boot for midplane IP assignment. In this example there are 8 DPUs and ach having a 200G interface.
+```
+"DPUs" : [
+    {
+      "dpu0": {
+                "midplane_interface":  "dpu0"
+       }
+    },
+    {
+       "dpu1": {
+                "midplane_interface":  "dpu1"
+        },
+    },
+    .
+    .
+    {
+       "dpuX": {
+                "midplane_interface":  "dpux"
+        },
+    },
+
+]
+```
+On the DPU's platform.json, we can have 
+```
+DPU: {
+
+  // Anything specific to DPU, else remain empty
+
+}
+```
+#### 3.1.4 ModuleBase class API enhancements
+get_base_mac(self):
+```
+    Retrieves the base MAC address for the module
+
+    Returns:
+        A string containing the MAC address in the format 'XX:XX:XX:XX:XX:XX'
+```
+
+get_system_eeprom_info(self):
+```
+    Retrieves the full content of system EEPROM information for the DPU module
+
+    Returns:
+        DPU modules return a dictionary with key value pairs of IDPROM 
+```
+
+get_name(self):
+```
+    Retrieves the name of the device
+
+    Returns:
+        string: The name of the device. Ex; SWITCH, DPU0, DPUX
+```
+
+get_description(self):
+```
+    Retrieves the platform vendor's product description of the module
+
+    Returns:
+        A string, providing the vendor's product description of the module.
+```
+
+get_slot(self):
```

Comment:
not applicable for smart switch

Reply:
Fixed the description: Retrieves the DPU ID in case of smartswitch

Reply:
@rameshraghupathy we discussed to remove slot concept from Smart swtich. Instead have new API get_dpu_id()